daemon/api/endpoints/flow.py

```diff
@@ -208,6 +205,7 @@ async def _ping(
     _, args, _ = ArgNamespace.get_parsed_args(kwargs, set_client_cli_parser())
     client = Client(args)
     try:
+        # TODO: this introduces side-effect, need to be refactored. (2020.01.10)
```

Comment:
`Client` & `WebsocketClient` need to be refactored to work as a context manager. This interface can tell the daemon client if the gateway server is alive.